src/BikeSharing.DomainLogic/BikeWorkout.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using BikeSharing.DomainLogic;
+
+namespace Training
+{
+    public class Workout
+    {
+        public DateTime Date { get; set; }
+        public TimeSpan Duration { get; set; }
+        public double AverageHeartRate { get; set; }
+        public string Notes;
+
+        public Workout(DateTime date, TimeSpan duration, double averageHeartRate, string notes)
+        {
+            Date = date;
+            Duration = duration;
+            AverageHeartRate = averageHeartRate;
+            if (notes == null)
+            {
+                throw new ArgumentNullException();
+            }
+            Notes = notes;
+        }
+    }
+
+    public class DistanceWorkout : Workout
+    {
+        public double Distance { get; set; }
+        public double Pace { get; }
+        public DistanceWorkout(double distance, DateTime datetime, TimeSpan duration, double rate, string notes) : base(datetime, duration, rate, notes)
+        {
+            Distance = distance;
+            Pace = distance / duration.TotalHours;
+        }
+    }
+
+    public class BikeWorkout : DistanceWorkout
+    {
+        public WorkoutType Type { get; set; }
+
+        public BikeWorkout(WorkoutType type, double distance, DateTime datetime, TimeSpan duration, double rate, string notes) : base(distance, datetime, duration, rate, notes)
+        {
+            Type = type;
+        }
+    }
+
+    public class Athlete
+    {
+        private readonly string[] HASHTAGS = new string[]{
+"#transformationtuesday",
+"#mcm",
+"#wcw",
+"#fitfam",
+"#fitspo",
+"#fitness",
+"#gymtime",
+"#treadmill",
+"#gainz",
+"#workout",
+"#getStrong",
+"#getfit",
+"#justdoit",
+"#youcandoit",
+"#bodybuilding",
+"#fitspiration",
+"#cardio",
+"#ripped",
+"#gym",
+"#geekabs",
+"#crossfit",
+"#beachbody",
+"#exercise",
+"#weightraining",
+"#training",
+"#shredded",
+"#abs",
+"#sixpacks",
+"#muscle",
+"#strong",
+"#lift",
+"#weights",
+"#Getfit",
+"#weightloss",
+"#wod",
+"#aesthetic",
+"#squad",
+"#shreadding",
+"#personaltrainer",
+"#dreambitviral",
+"#quote",
+"#quotes",
+"#inspiring",
+"#motivation",
+"#fitnessquote",
+"#youcandoit",
+"#justbringit",
+"#dreambig",
+"#success",
+"#staypositive",
+"#noexcuses"
+    };
+
+        public string Username { get; set; }
+        public Gender Gender { get; set; }
+        public int Age { get; set; }
+        public double Weight { get; set; }
+        public double Height { get; set; }
+        public double BasalMetabolicRate { get; }
+        public List<Workout> Workouts { get; set; }
+
+        public Athlete(string username, Gender gender, int age, double weight, double height)
+        {
+            Username = username;
+            Gender = gender;
+            Age = age;
+            Weight = weight;
+            Height = height;
+            BasalMetabolicRate = GetBasalMetabolicRate();
```

Comment:
why does this need to be stored in a field?  Why not just have the property compute the result when asked?